### PR TITLE
Update to master

### PR DESCRIPTION
**Pull request by vhbit**
- Sized? reform
- String/Show split
- indexing
- features reform

### CODE REVIEW

**Comment by alexcrichton** [comment]
The current master should be compatible with the current nightly, but I'll merge this once we've got new nightlies.

File: src/json.rs [hunk 1]
```diff
@@ -414,7 +413,7 @@ fn spaces(wr: &mut fmt::Writer, mut n: uint) -> fmt::Result {
     }
 
     if n > 0 {
-        wr.write_str(BUF[..n])
+        wr.write_str(BUF.index(&(..n)))
```
Comment (oli-obk) [inline src/json.rs]: i think the previous version works again now

Comment (vhbit) [inline src/json.rs]: May be, haven't checked yet.

**Comment by vhbit** [comment]
@oli-obk Do you plan to update it to current master?

**Comment by vhbit** [comment]
Partially updated to master - test do not pass yet. Also disabled treating warnings as errors for tests (as there is a lot of staging warnings)

**Comment by vhbit** [comment]
Updated doc tests too

File: src/json.rs (hunk 1 above)
Comment (vhbit) [inline src/json.rs]: The fun thing was it worked partially as changed from `buf[..n]` to `&buf[..n]`

**Comment by steveklabnik** [comment]
Building this still fails for me with 

```
   Compiling rustc-serialize v0.2.6 (https://github.com/vhbit/rustc-serialize?ref=up-master#2cb0aae1)
/home/steve/.cargo/git/checkouts/rustc-serialize-7fb0fdffe3ea5e99/up-master/src/json.rs:1250:52: 1250:54 error: expected one of `(`, `::`, `]`, or an operator, found `..`
/home/steve/.cargo/git/checkouts/rustc-serialize-7fb0fdffe3ea5e99/up-master/src/json.rs:1250                     &self.str_buffer[start as uint .. start as uint + size as uint]).unwrap())
                                                                                                                                                ^~
Could not compile `rustc-serialize`.

```

**Comment by alexcrichton** [comment]
Rolled into https://github.com/rust-lang/rustc-serialize/pull/21 with extra updates for the alpha release.

**Pull request closed without merging**